### PR TITLE
HTMLCleaner doesn't recognize some HTML named entity

### PR DESCRIPTION
**Pull request by surli**
- Fix the pattern to recognize entities
  - Add one more test

### CODE REVIEW

**Comment by surli** [comment]
@tmortagne this is ready. I checked on the HTML4 RFC https://www.w3.org/TR/html401/sgml/entities.html and following wikipedia pages: https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references I think we cover all cases. If you've a better source to ensure the pattern tell me.

File: xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java [hunk 1]
```diff
@@ -73,7 +73,7 @@
         /**
          * Regex to recognize a XML Entity.
          */
-        private static final Pattern ENTITY = Pattern.compile("&[a-zA-Z]+;|&#[0-9a-zA-Z]+;");
+        private static final Pattern ENTITY = Pattern.compile("&[a-zA-Z0-9]+;|&#[0-9a-zA-Z]+;");
```
Comment (vmassol) [inline xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java]: A documentation link to where this is coming from would be great. Isn't there some grammar that exists somewhere?

Comment (surli) [reply xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java]: Didn't find it yet. If you have an idea where to find it I take it.

Comment (vmassol) [reply xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java]: https://www.w3.org/TR/WD-xml-lang-970630#sec4.3 ?

Comment (vmassol) [reply xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java]: So for example `%_` seems valid but you don't seem to have it in your regex...

Comment (vmassol) [reply xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java]: Next question: is there some existing regex or API to check that?

Comment (surli) [reply xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java]: > My point is that we need to check against the grammar to be sure to not forget anything. 

Yes I agree that was part of my question when posting the sources :)

**Comment by vmassol** [comment]
Note: I haven't checked the grammar so I'm trusting you :)

**Pull request merged**